thetatauCMT/chapters/models.py

```diff
@@ -516,7 +518,121 @@ def get_generic_chapter_emails(self):
+            if no_previous == False: #  in the for loop would not work
+                #loops for the number of officers (5 officer positions as of Dec 2021)
+                #checks if current and future officer at a specific index is true
+                #if true filter the list at index and checks if query set has a officer set to expire after 14 days labeled future
+                #if future is not true then the current officer needs to be notified and that position needs to be updates
+                #if current_and_future_officer[officers] is false then that position is et to be updated and the previous officer will be notified
+                for officers in range(len(pos)): 
+                    if current_and_future_officer[officers]: 
+                        future = current_and_future_officer[officers].filter(roles__end__gte=TODAY_END + timedelta(14),) #gets the next true
+                        if not future:
+                            current = current_and_future_officer[officers].first()
+                            officers_to_update.append(pos[officers])
+                            officers_to_notify.append(current)
+                    else:
+                        officers_to_update.append(pos[officers])
+                        officers_to_notify.append(previous_officer[officers])
+
+                #gets the email of the user that we need to Notify
+                emails_from_user = [user.email for user in officers_to_notify]
+                #adds emails from user to the email list already used
+                emails.extend(emails_from_user)
+                    
+                #gets all the positions that need to change and formats 
+                #it to one string to use as a varaible in the html
+                for officers in range(len(officers_to_notify)):
+                    end = len(officers_to_notify) - 1
+                    if end == officers:
+                        html_officers = html_officers + " and " + str(officers_to_update[officers])
+                    else:
+                        html_officers = html_officers + str(officers_to_update[officers]) + ", "
+
+                #gets all the positions that need to change and formats 
+                #it to one string to use as a varaible in the html
+                for officers in range(len(officers_to_notify)):
+                    end = len(officers_to_notify) - 1
+                    if end == officers:
+                        html_members = html_members + " and " + str(officers_to_notify[officers])
+                    else:
+                        html_members = html_members +  str(officers_to_notify[officers]) + ", "
+            else:#for all empty chapter sends email about all positions to just the chapter email
+                for officers in pos:
+                    officers_to_notify.append(officers)
+                while("" in emails) :
```

Comment:
I've never needed a while loop, search all code in repo, never used one. 
You can remove missing with a simple list comprehension 
[email for email in emails if email]